x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/save_custom_dashboard.ts

```diff
@@ -37,10 +40,18 @@ export function initSaveCustomDashboardRoute(framework: KibanaFramework) {
 
       await checkCustomDashboardsEnabled(uiSettingsClient);
 
-      const payload: InfraSaveCustomDashboardsRequestPayload = request.body;
+      const payload: InfraSaveCustomDashboardsResponseBody = {
+        ...request.body,
+        assetType: request.params.assetType,
+      };
       const customDashboards = await findCustomDashboard(payload.assetType, savedObjectsClient);
```

Comment:
We're checking the existence of a saved object in the line below, getting this total here might not be needed anymore